galsim/catalog.py

```diff
@@ -312,3 +317,334 @@ def itervalues(self):
+        bulge_beta = params[15]*galsim.radians
+        bulge_hlr = 0.03*np.sqrt(bulge_q)*params[9]
+        bulge_flux = 2.0*np.pi*3.607*(bulge_hlr**2)*params[8]/0.03**2
+        # Disk parameters.
+        disk_q = params[3]
+        disk_beta = params[7]*galsim.radians
+        disk_hlr = 0.03*np.sqrt(disk_q)*params[1]
+        disk_flux = 2.0*np.pi*1.901*(disk_hlr**2)*params[0]/0.03**2
+        bfrac = bulge_flux/(bulge_flux+disk_flux)
+        # Make sure the bulge-to-total flux ratio is not nonsense.
+        if bfrac < 0 or bfrac > 1 or np.isnan(bfrac):
+            raise RuntimeError("Cannot make parametric model for this object")
+
+        # Then make the object.
+        if chromatic:
+            bulge = galsim.DeVaucouleurs(flux=1., half_light_radius=record['size_factor']*bulge_hlr) \
+                * sed_bulge.atRedshift(record['zphot']).withMagnitude(
+                record['mag_auto']-2.5*math.log10(bfrac*record['flux_factor']), _COSMOS_bandpass)
+            disk = galsim.Exponential(flux=1., half_light_radius=record['size_factor']*disk_hlr) \
+                * sed_disk.atRedshift(record['zphot']).withMagnitude(
+                record['mag_auto']-2.5*math.log10((1.-bfrac)*record['flux_factor']), _COSMOS_bandpass)
+        else:
+            bulge = galsim.DeVaucouleurs(flux = record['flux_factor']*bulge_flux,
+                                         half_light_radius = record['size_factor']*bulge_hlr)
+            disk = galsim.Exponential(flux = record['flux_factor']*disk_flux,
+                                      half_light_radius = record['size_factor']*disk_hlr)
+        # Apply shears for intrinsic shape.
+        if bulge_q < 1.:
+            bulge = bulge.shear(q=bulge_q, beta=bulge_beta)
+        if disk_q < 1.:
+            disk = disk.shear(q=disk_q, beta=disk_beta)
+        return bulge+disk
+    else:
+        (fit_gal_flux, fit_gal_hlr, fit_gal_n, fit_gal_q, _, _, _, fit_gal_beta) = sparams
+
+        gal_n = fit_gal_n
+        # Fudge this if it is at the edge of the allowed n values.  Since GalSim (as of #325 and
+        # #449) allow Sersic n in the range 0.3<=n<=6, the only problem is that the fits
+        # occasionally go as low as n=0.2.
+        if gal_n < 0.3: gal_n = 0.3
+        gal_q = fit_gal_q
+        gal_beta = fit_gal_beta*galsim.radians
+        gal_hlr = 0.03*np.sqrt(gal_q)*fit_gal_hlr
+        # Below is the calculation of the full Sersic n-dependent quantity that goes into the
+        # conversion from surface brightness to flux, which here we're calling 'prefactor'.  In the
+        # n=4 and n=1 cases above, this was precomputed, but here we have to calculate for each
+        # value of n.
+        tmp_ser = galsim.Sersic(gal_n, half_light_radius=1.)
+        gal_bn = (1./tmp_ser.getScaleRadius())**(1./gal_n)
+        prefactor = gal_n * _gammafn(2.*gal_n) * math.exp(gal_bn) / (gal_bn**(2.*gal_n))
+        gal_flux = 2.*np.pi*prefactor*(gal_hlr**2)*fit_gal_flux/0.03**2
```

Comment:
Sadly, that would remove the need for the Rosetta Code gamma function.  But perhaps that tidbit could be moved to galsim.utilities.py in case we have further use of it.